umap/static/umap/js/modules/saving.js

```diff
@@ -0,0 +1,54 @@
+export class SaveManager {
+  constructor() {
+    this._queue = new Set()
+  }
+
+  get isDirty() {
+    return Boolean(this._queue.size)
+  }
+
+  async save() {
+    for (const obj of this._queue) {
+      const ok = await obj.save()
+      if (!ok) break
+      this.delete(obj)
+    }
+  }
+
+  add(obj) {
+    this._queue.add(obj)
+    this.checkStatus()
+  }
+
+  delete(obj) {
+    this._queue.delete(obj)
+    this.checkStatus()
+  }
+
+  has(obj) {
+    return this._queue.has(obj)
+  }
+
+  checkStatus() {
```

Comment:
I'm not sure I get the difference between `checkStatus` and `isDirty` concepts, can't we merge them both?

Reply:
Bad naming then I guess :)

`isDirty` is about a given object to need to be saved to the server or no.
`checkStatus` is the SaveManager that update the UI to reflect that there is at least one object to be saved to the server.

Reply:
Would `updateDOM` be a better naming here (instead of `checkStatus`, which says about `how` but not about `what`) ?